sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernatePoolDao.java

```diff
@@ -143,40 +160,104 @@ public Session currentSession() {
   }
 
   @Override
+  public Pool get(long poolId) throws IOException {
+    return withWatcherGroup((PoolImpl) currentSession().get(PoolImpl.class, poolId));
+  }
+
+  public BoxStore getBoxStore() {
+    return boxStore;
+  }
+
+  @Override
+  public Pool getByBarcode(String barcode) throws IOException {
+    if (barcode == null) throw new NullPointerException("cannot look up null barcode");
+    // This is not guaranteed to be unique, so we pick one at pseudo-random.
```

Comment:
Okay. The tests believe otherwise and H2 is fine with there being duplicates despite their being a unique key in mySQL. o_O